core/arch/arm/kernel/delay.c

```diff
@@ -44,3 +48,37 @@ void mdelay(uint32_t ms)
 {
 	udelay(1000 * ms);
 }
+
+uint64_t utimeout_init(uint32_t us)
+{
+	return read_cntpct() + US2CNT(us);
+}
+
+bool utimeout_elapsed(uint32_t us, uint64_t reference)
+{
+	uint64_t origin = reference - US2CNT(us);
+	uint64_t now = read_cntpct();
+
+	if (origin < reference)
+		return now < origin || now > reference;
+
+	return now < origin && now > reference;
+}
+
+uint32_t utimeout_elapsed_us(uint32_t us, uint64_t reference)
```

Comment:
Could be simplified too. Perhaps it would be better not to introduce this function now since it is not used for the moment.

Reply:
indeed, this is useful for debugging only :)